ComputerVision/cannyVariables.py

```diff
@@ -0,0 +1,120 @@
+    if settings.thresholdVariables["currentlt"] >= settings.thresholdVariables["currentht"] :
+        settings.thresholdVariables["currentht"] = settings.thresholdVariables["currentlt"] + 1
+        settings.sliders["highthresholdslider"].set(settings.thresholdVariables["currentht"])
+    gausimg = cv2.GaussianBlur(settings.images["originalphotowithCV"], (5,5), settings.thresholdVariables["currentsigma"])
+    edges = cv2.Canny(gausimg, settings.thresholdVariables["currentlt"], settings.thresholdVariables["currentht"])
+    edges = Image.fromarray(edges)
+    edges = ImageTk.PhotoImage(image = edges)
+    settings.imagelabels["altered"].configure(image=edges)
+    settings.imagelabels["altered"].image = edges
+    settings.images["edges"] = edges
+    
+
+def change_high_threshold (ht) :
+    settings.thresholdVariables["currentht"] = int(ht)
+    if settings.thresholdVariables["currentht"] <= settings.thresholdVariables["currentlt"] :
+        settings.thresholdVariables["currentlt"] = settings.thresholdVariables["currentht"] - 1
+        settings.sliders["lowthresholdslider"].set(settings.thresholdVariables["currentlt"])
+    gausimg = cv2.GaussianBlur(settings.images["originalphotowithCV"], (5,5), settings.thresholdVariables["currentsigma"])
+    edges = cv2.Canny(gausimg, settings.thresholdVariables["currentlt"], settings.thresholdVariables["currentht"])
+    edges = Image.fromarray(edges)
+    edges = ImageTk.PhotoImage(image = edges)
+    settings.imagelabels["altered"].configure(image=edges)
+    settings.imagelabels["altered"].image = edges
+    settings.images["edges"] = edges
+   
+
+def images () :
+    photo = tk.PhotoImage(file = settings.imagepath)
+    photo = photo.subsample(2,2)
+
+    image = cv2.imread(settings.imagepath)
+    dim = (int(image.shape[1] * 50 / 100), int(image.shape[0] * 50 / 100))
+    image = cv2.resize(image, dim, interpolation = cv2.INTER_LINEAR)
+    image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+
+    edges = cv2.Canny(cv2.GaussianBlur(image, (5,5), settings.thresholdVariables["currentsigma"]),settings.thresholdVariables["currentlt"],settings.thresholdVariables["currentht"])
+    edges = Image.fromarray(edges)
+    edges = ImageTk.PhotoImage(image = edges)
+
+    settings.images["originalphoto"] = photo
+    settings.images["originalphotowithCV"] = image
+    settings.images["edges"] = edges
+
+def frames () :
+    settings.frames["imageframe"] = tk.Frame(master=settings.m,background="white")
+    settings.frames["originalimage"] = tk.Frame(master=settings.frames["imageframe"], background="white", borderwidth=1)
+    settings.frames["alteredimage"] = tk.Frame(master=settings.frames["imageframe"], background="white", borderwidth=1)
+
+    settings.frames["sliderframe"] = tk.Frame(master=settings.m, background="white")
+
+    settings.frames["sigmasliderframe"] = tk.Frame(master=settings.frames["sliderframe"], background="white", relief=tk.RAISED, borderwidth=1, width=50, height=50)
```

Comment:
i would extract the constants into a separate file, "sliderframe" and imageframe for example, it may not be necessary ofc if you only have it in one place and dont plan to use it elsewhere

Reply:
If the variable is used between files, it is in a global variable inside the settings file, so if they arent, they are only within that one file xD if it changes, ill be sure to put it in the settings file

But yes! I agree